base/range.jl

```diff
@@ -415,6 +415,13 @@ end
 
 function getindex{T}(v::Range{T}, i::Integer)
     @_inline_meta
+    ret = convert(T, first(v) + i - 1)
+    @boundscheck ((i > 0) & (ret <= v.stop)) || throw_boundserror(v, i)
```

Comment:
> Is there an edge case where ret overflowed?

First, note this is a checked conversion, so for any type strictly smaller than `Int` there is no problem.

Initially I didn't think there was a problem even for `Int`, but now I see one: `r = typemax(Int)-5:typemax(Int)-1; r[7]`. (For the record, this is a completely stupid corner case that no one should actually care about.) I'll see if I can fix it; thanks for prompting me to think a little more carefully!

> Should that & be a &&?

Nope. Otherwise you end up with two branches instead of one, and it's about 40% slower.